### PR TITLE
[Xamarin.Android.Build.Tests] Add support for testing on system applications.

### PR DESCRIPTION
**Pull request by dellis1972**
Context https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1288717

The new fast deployment system broke fast deployment for system installed
applications. This commit adds unit tests to stop this kind of regression
in the future.

In order to install a `system` application we need a few things to happen

1. The emulator MUST be started with the `-writable-system` argument
2. The apk needs to be signed with a platform keystore.
3. The AndroidManifest must include `android:sharedUserId="android.uid.system"`
   in its `manifest` element.

- [x] Figure out a way to sign with the `platform` keys at https://github.com/aosp-mirror/platform_build/tree/master/target/product/security
- [x] Include https://github.com/xamarin/monodroid/pull/1177

### CODE REVIEW

**Comment by dellis1972** [comment]
The nightly build has the new test and it passed https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4543632&view=results

**Comment by dellis1972** [comment]
Some tests are failing because the emulator is disappearing half way through the test run... weird. 
Trying to run the tests again.

File: src/Xamarin.Android.Build.Tasks/Properties/Resources.resx [hunk 1]
```diff
@@ -712,8 +712,8 @@ The following are literal names and should not be translated: ABI, 'libs/armeabi
     <comment>The following are literal names and should not be translated: MultiDexMainDexList</comment>
   </data>
   <data name="XA4310" xml:space="preserve">
-    <value>`$(AndroidSigningKeyStore)` file `{0}` could not be found.</value>
-    <comment>The following are literal names and should not be translated: `$(AndroidSigningKeyStore)`</comment>
+    <value>`{0}` file `{1}` could not be found.</value>
+    <comment>{0} - The MSBuildProperty which contains the file name value.</comment>
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Properties/Resources.resx:716]: I think this needs to say what `{1}` is.

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs [hunk 2]
```diff
@@ -312,5 +313,15 @@ protected static bool SwitchUser (string username)
...
+
+		protected byte [] GetKeystore (string keyname = "test.keystore")
+		{
+			var assembly = typeof (XamarinAndroidCommonProject).Assembly;
+			using (var stream = assembly.GetManifestResourceStream ($"Xamarin.ProjectTools.Resources.Base.{keyname}")) {
+				var data = new byte [stream.Length];
+				stream.Read (data, 0, (int) stream.Length);
+				return data;
+			}
+		}
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/DeviceTest.cs:325]: We started a new place for data like this, should we move this here:

https://github.com/xamarin/xamarin-android/blob/main/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/ResourceData.cs

**Pull request merged**